Mysterious Build Error on OS X:  default initialization of an object of const type Foo without a user-provided default constructor

I am using 839b6e017bba05f92aaf6dc95f308c374f99d333 on OSX and building using `bazel`.

# The Error

```
ERROR: /Users/liang/dev/drake-distro-3/drake/systems/framework/BUILD:491:1: C++ compilation of rule '//drake/systems/framework:value_checker_test' failed: sandbox-exec failed: error executing command /usr/bin/sandbox-exec -f /private/var/tmp/_bazel_liang/3302fbe3d4d2eae8487460393c40acaf/bazel-sandbox/8bbfb44b-02d0-4354-8ae2-f5e5a3578d43-7/sandbox.sb ... (remaining 70 argument(s) skipped).
drake/systems/framework/test/value_checker_test.cc:66:25: error: default initialization of an object of const type 'const drake::systems::(anonymous namespace)::NotSoGoodVector' without a user-provided default constructor
  const NotSoGoodVector not_so_good;
                        ^
                                   {}
drake/systems/framework/test/value_checker_test.cc:67:28: error: default initialization of an object of const type 'const drake::systems::(anonymous namespace)::NotQuiteGoodVector' without a user-provided default constructor
  const NotQuiteGoodVector not_quite_good;
                           ^
                                         {}
2 errors generated.
```

--- a/drake/systems/framework/test/value_checker_test.cc
+++ b/drake/systems/framework/test/value_checker_test.cc
@@ -39,14 +39,14 @@ class MissingCloneVector : public BasicVector<double> {
 class NotSoGoodVector : public GoodVector {
  public:
   DRAKE_NO_COPY_NO_MOVE_NO_ASSIGN(NotSoGoodVector)
-  NotSoGoodVector() = default;
+  NotSoGoodVector() {}
 };
 
 // This one remembers to DoClone in the subclass, but does it wrong.
 class NotQuiteGoodVector : public GoodVector {
  public:
   DRAKE_NO_COPY_NO_MOVE_NO_ASSIGN(NotQuiteGoodVector)
-  NotQuiteGoodVector() = default;
+  NotQuiteGoodVector() {}
   GoodVector* DoClone() const override {
     auto result = new GoodVector;
     result->set_value(this->get_value());

